### DISCUSSION TITLE
Barcode Reading direction

### DISCUSSION THREAD

**Discussion by sunil-kumar2907** [micjahn/ZXing.Net#417] [Q&A]
Hello Everyone,

I am working on a barcode detection project. I want to start reading either from left side of image or right side of image. By default start reading reading from center of the imege. Anyone have any idea reagdring how change the reading direction using Zxing/C# in windows application. 

Thank you in advance.

**Comment by micjahn** [comment]
There is no easy answer to your question.
The reader implementation for one-dimensional barcodes is different from the two-dimensional ones like QR or DataMatrix.
If you want to change the 1D scanning you have to change the implementation of the doDecode method in the class OneDReader.
For the 2D codes, you have to change the different finder implementations like ZXing.QrCode.Internal.FinderPatternFinder and ZXing.Datamatrix.Internal.Detector.

**Reply by sunil-kumar2907** [reply]
Thank you Michael for your guidance.
